invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js

```diff
@@ -256,18 +258,124 @@ class CommunityProfileForm extends Component {
       },
     });
 
+    /**
+     * Deserializes a funding record (e.g. funder or award)
+     *
+     * @param {object} fund
+     *
+     * @returns {object} an object containing the deserialized record
+     */
+     const deserializeFunding = ((fund) => {
+      const _deserialize = ((value) => {
+        const deserializedValue = _cloneDeep(value);
+
+        if (value?.title?.en) {
+          deserializedValue.title = value.title.en;
+        }
+
+        if (value.identifiers) {
+          const allowedIdentifiers = ['url'];
+
+          allowedIdentifiers.forEach((identifier) => {
+            let identifierValue = null;
+            value.identifiers.forEach((v) => {
+              if (v.scheme === identifier) {
+                identifierValue = v.identifier;
+              }
+            });
+
+            if (identifierValue) {
+              deserializedValue[identifier] = identifierValue;
+            }
+          })
+
+          delete deserializedValue['identifiers'];
+        }
+        return deserializedValue;
+      });
+
+      let deserializedValue = {};
+      if (fund !== null) {
+        deserializedValue = Array.isArray(fund)
+          ? fund.map(_deserialize)
+          : _deserialize(fund);
+      }
+
+      return deserializedValue;
+    });
+
     // Deserialize organizations
     let organizations = _map(
       _get(initialValues, "metadata.organizations"),
       "name"
     );
+
+    // Deerialize each funding record, award being optional.
+    const funding = initialValues.metadata.funding.map((fund) => {
+      return {
+        ...(fund.award && {award: deserializeFunding(fund.award)}),
+        funder: deserializeFunding(fund.funder)
+      }
+    });
+
     return {
       ...initialValues,
-      metadata: { ...initialValues.metadata, organizations },
+      metadata: { ...initialValues.metadata, organizations, funding },
     };
   };
 
+  /**
+   * Serializes community values
+   *
+   * @param {object} values
+   *
+   * @returns
+   */
   serializeValues = (values) => {
+    /**
+     * Serializes a funding record (e.g. funder or award)
+     *
+     * @param {object} fund
+     *
+     * @returns {object} an object containing the serialized record
+     */
+     const serializeFunding = ((fund) => {
+      const _serialize = (value) => {
```

Comment:
This is exactly the same logic to serialise a funding record that was used in

https://github.com/inveniosoftware/react-invenio-deposit/blob/aefebaa2156c752ee10829d69c85b3ab22a15eeb/src/lib/fields/VocabularyField.js#L296

Reply:
Can you just import the field and use the serialize method? Maybe it's not exported but we could export the field eventually....